### PR TITLE
Add options to let the puppet master listen for HTTP connections too

### PR DESCRIPTION
**Pull request by whefter**
(load balancer/reverse proxy scenarios), protected by default Deny all. Configurable listen port and allow directive.

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -171,6 +171,19 @@
...
+#                                  the HTTP puppet master denies access from all clients by default,
+#                                  allowed clients must be specified with $server_http_allow.
+#                                  type:boolean
+#
+# $server_http_port::              Puppet master HTTP port; defaults to 8139.
+#                                  type:integer
+#
+# $server_http_allow::             Array of allowed clients for the HTTP puppet master. Passed
+#                                  to Apache's 'Allow' directive.
+#                                  type:integer
```
Comment (mmoll) [inline manifests/init.pp]: should probably be type:array

**Comment by mmoll** [comment]
Tested, works for me. :+1: 
@domcleal? @ekohl?

**Comment by whefter** [comment]
Oops, yeah. Corrected that, and rebased/squashed the commits into one.

File: manifests/server/passenger.pp [hunk 2]
```diff
@@ -87,4 +92,31 @@
...
+            default => $custom_fragment
+          },
+          'SetEnvIf X-Client-Verify "(.*)" SSL_CLIENT_VERIFY=$1',
+          'SetEnvIf X-SSL-Client-DN "(.*)" SSL_CLIENT_S_DN=$1',
+        ], "\n"),
+      options         => ['None'],
+      require         => Class['::puppet::server::rack'],
+    }
+  }
+}
```
Comment (ekohl) [inline manifests/server/passenger.pp]: nitpick: I'd prefer a newline at the end of the file.

**Comment by ekohl** [comment]
I'm unsure. My main concern is that beginners might try this. I do see the use case so maybe it could be documented in the README as an advanced scenario?

**Comment by mmoll** [comment]
More documentation is always good :memo:

**Comment by mmoll** [comment]
:+1:

**Comment by ekohl** [comment]
:+1: but I'd like @domcleal to take a look as well.

**Comment by mmoll** [comment]
merged as also @domcleal's feedback via IRC was positive, thanks @whefter! :tada:

**Pull request merged**